### PR TITLE
refactor : improved folder structure and linter setup

### LINKED ISSUE #8: Reorganise Folder structure
This discussion has bee started to know if we can organise the folder in better way and hence refactor code 

kapp-api/server/server.js  [ entery point ]
kapp-api/server/middlerware [ for all middle ware authentication and other things]
kapp-api/server/models. [all schema designs ]
kapp-api/server/controllers [ all actual functionalities ] 
kapp-api/server/routes [ all routes ] 
 
Also IG we should break it into functions
```
addArticle = (article) => {
try {
     const newArticle = await article.save();
  }
catch (err) {
  throw "err";
}
}
```
```
router.post('/', async (req,res) => {
    try {
        NewArticle = addArticle({req.body.title,req.body.description,req.body.author,req.body.tags})
        //201 cause its sucess
        res.status(201).json(newArticle)
    } catch(arr) {
        res.status(400).json( { message : err.message })
    }
})
```

Why the above method instead of now 

- Easy to navigate which function is called when post method and when get method
- Suppose if get method and post method follows some common preprocessing steps that can be shared ( I am assuming serialisation would could come in picture )
- We can do unit testing of the functions , that it is handling correct data types and everything
- We can do similar testing if api response is correct
- If we follow current approach then we don't need to write functional testing because we don't have any functions and then we need to cover everything in api testing, if the error msg is nice or not, if the articles has been deleted or article not there and all this 
Which I guess is not correct 
- I guess api testing should be , that if I hit the api, I just get response and every-time the response should have message property, I don't care about what the message is or why the failure.
- Functional testing is , If I supply this data then then message should be `success` and if I supply other data then the `message` should be `failed`

**For more explanation  Please visit** < https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/routes >

### PR DESCRIPTION
**Pull request by axiiomatic**
resolve #8 

Also added eslint and prettier config.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "arrowParens": "avoid",
+  "tabWidth": 2,
+  "semi": true,
+  "singleQuote": true
+}
```
Comment (jeffin143) [inline .prettierrc:6]: Thanks for this too

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  parserOptions: {
+    ecmaVersion: 2018,
+  },
+  rules: {
+    'linebreak-style': ['error', 'unix'],
+    quotes: ['error', 'single'],
+    semi: ['error', 'always'],
+  },
+};
```
Comment (jeffin143) [inline .eslintrc.js:20]: Thanks for setting up Linting 🎉

File: package.json [hunk 3]
```diff
@@ -4,7 +4,9 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "devStart": "nodemon server.js",
+    "devStart": "nodemon server/server.js",
+    "lint:fix": "eslint --fix ./server",
+    "prettier": "prettier ./server --write",
```
Comment (jeffin143) [inline package.json:9]: add two more for just checking lint, and avoid fixing
so 
```
"lint" : "eslint  ./server"
```
similarly for other too

File: server/controllers/articleController.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// controller for articles endpoint
+
+module.exports = {
+  // addArticle: (req, res) => {},
+};
```
Comment (jeffin143) [inline server/controllers/articleController.js:5]: Logic should be here 
Screenshot 2020-04-25 at 6 59 45 PM

File: server/routes/articles.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    const articles = await Article.find();
+
+    //since for now there is no search option
+    //our req is always sucess so 201
+    res.status(201).json(articles);
+  } catch (err) {
+    //500 for any internal error i.e my fault
+    res.status(500).json({ message: err.message });
+  }
+});
```
Comment (jeffin143) [inline server/routes/articles.js:35]: Only function calls should be there, Logic should be removed 
Screenshot 2020-04-25 at 6 59 57 PM

**Comment by axiiomatic** [comment]
Messed up. will make a new PR.

**Pull request closed without merging**